### PR TITLE
ts: adds support for manual backup of tokens to file

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- support for backup of tokens on a cluster to a file

## Why are we making these changes?

It allows us to maintain a manual backup of all tokens on a specific cluster.

## Example output 

```
$ lein run -- -a -b data.json http://localhost:9091
2018-03-05 14:43:24 INFO  token-syncer.main - command-line arguments: [-a -b data.json http://localhost:9091]
2018-03-05 14:43:24 INFO  eclipse.jetty.util.log - Logging initialized @7303ms to org.eclipse.jetty.util.log.Slf4jLog
2018-03-05 14:43:24 INFO  token-syncer.main - backing up tokens to data.json in accretion mode 
2018-03-05 14:43:24 INFO  token-syncer.backup - reading contents from data.json
2018-03-05 14:43:24 INFO  token-syncer.backup - found 2 tokens in data.json : (token0AAA.testtokencreatedeleteshamsimam689423.127.0.0.1 token1.testtokencreatedeleteshamsimam689423.127.0.0.1)
2018-03-05 14:43:24 INFO  token-syncer.waiter - token-syncer.101bed2f-25bc-4a67-ade1-16c155a17732 making GET request to http://localhost:9091/tokens
2018-03-05 14:43:25 INFO  token-syncer.backup - found 10 tokens on http://localhost:9091 : #{token0.testtokencreatedeleteshamsimam689423.127.0.0.1 token1.testtokencreatedeleteshamsimam689423.127.0.0.1 token2.testtokencreatedeleteshamsimam689423.127.0.0.1 token3.testtokencreatedeleteshamsimam689423.127.0.0.1 token4.testtokencreatedeleteshamsimam689423.127.0.0.1 token5.testtokencreatedeleteshamsimam689423.127.0.0.1 tok
en6.testtokencreatedeleteshamsimam689423.127.0.0.1 token7.testtokencreatedeleteshamsimam689423.127.0.0.1 token8.testtokencreatedeleteshamsimam689423.127.0.0.1 token9.testtokencreatedeleteshamsimam689423.127.0.0.1}
2018-03-05 14:43:25 INFO  token-syncer.waiter - token-syncer.c188ddf6-91f5-4fb2-97f1-695a1ec97897 making GET request to http://localhost:9091/token
2018-03-05 14:43:25 INFO  token-syncer.waiter - loading token0.testtokencreatedeleteshamsimam689423.127.0.0.1 on http://localhost:9091 responded with status 200 {content-type application/json, etag E-c4baf3b3e3f5af1cc19d344c2993339f, x-cid token-syncer.53e7ff81-b734-4baf-8bd0-59e263f1c038}
...
2018-03-05 14:43:25 INFO  token-syncer.waiter - token-syncer.290fdca2-9581-4755-8086-e0d70c12445f making GET request to http://localhost:9091/token
2018-03-05 14:43:25 INFO  token-syncer.waiter - loading token9.testtokencreatedeleteshamsimam689423.127.0.0.1 on http://localhost:9091 responded with status 200 {content-type application/json, etag E-ff589dd48e1b021d0a484e3b0e4416b2, x-cid token-syncer.d7dd2663-9c60-4bd1-ae1b-8e26840d51f4}
2018-03-05 14:43:25 INFO  token-syncer.backup - writing 11 tokens to data.json : (token0.testtokencreatedeleteshamsimam689423.127.0.0.1 token0AAA.testtokencreatedeleteshamsimam689423.127.0.0.1 token1.testtokencreatedeleteshamsimam689423.127.0.0.1 token2.testtokencreatedeleteshamsimam689423.127.0.0.1 token3.testtokencreatedeleteshamsimam689423.127.0.0.1 token4.testtokencreatedeleteshamsimam689423.127.0.0.1 token5.test
tokencreatedeleteshamsimam689423.127.0.0.1 token6.testtokencreatedeleteshamsimam689423.127.0.0.1 token7.testtokencreatedeleteshamsimam689423.127.0.0.1 token8.testtokencreatedeleteshamsimam689423.127.0.0.1 token9.testtokencreatedeleteshamsimam689423.127.0.0.1)
2018-03-05 14:43:25 INFO  token-syncer.main - exiting with code 0
```

### CODE REVIEW

File: token-syncer/src/token_syncer/main.clj [hunk 1]
```diff
@@ -37,7 +38,12 @@
 (defn parse-cli-options
   "Parses and returns the cli options passed to the token syncer."
   [args]
-  (->> [["-d" "--dry-run"
+  (->> [["-a" "--backup-accrete"
```
Comment (sradack) [inline token-syncer/src/token_syncer/main.clj]: Why have an option if it's the default mode?

File: token-syncer/src/token_syncer/main.clj [hunk 2]
```diff
@@ -37,7 +38,12 @@
...
+         "Accrete new token descriptions into the backup file if it exists instead of overwriting (default behavior) the file."]
+        ["-b" "--backup-file file"
```
Comment (sradack) [inline token-syncer/src/token_syncer/main.clj]: Does the CLI library support subcommands?  Might be nice to have separate `sync` and `backup` commands, with `--accrete` (if necessary) an option only supported for `backup`.

Comment (sradack) [reply token-syncer/src/token_syncer/main.clj]: What does it mean to have a `--dry-run` of a backup operation?  Perhaps it's only useful for `sync`.

Comment (sradack) [reply token-syncer/src/token_syncer/main.clj]: Should we take this opportunity to rename this utility from `token-syncer` to `waiter-admin`

**Comment by sradack** [comment]
Please rebase.

**Comment by shamsimam** [comment]
@sradack Rebased.

**Pull request merged**